scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1932,19 +1941,23 @@ private void sendMessage(final long nanos, final RawData message, final Connecti
 		}
 		LOGGER.debug("Sending application layer message to [{}]", message.getEndpointContext());
 
+		String handshakeMode = message.getEndpointContext().get(DtlsEndpointContext.KEY_HANDSHAKE_MODE);
 		DTLSSession session = connection.getEstablishedSession();
 		SessionTicket ticket = connection.getSessionTicket();
 		if (session == null && ticket == null) {
-			if (serverOnly) {
-				message.onError(new EndpointUnconnectedException("server only, connection missing!"));
-				return;
-			}
```

Comment:
This means, in certain condition we refuse some some rawData instead of deferred it ?
That was a but ?

Reply:
In my setup, I use "server only" for "servers only". The nature there is, that it sends only responses. And with that, it's very probable, that the response will be rejected by the matcher after the handshake succeeded. So for my use-cases this doesn't change.